Model/AdyenInitPayments.php

```diff
@@ -160,21 +159,20 @@ public function execute(int $adyenCartId, string $stateData): string
      */
     protected function buildPaymentsRequest(Quote $quote, array $stateData): array
     {
-        $chargedQuoteCurrency = $this->chargedCurrency->getQuoteAmountCurrency($quote);
-        $currency = $chargedQuoteCurrency->getCurrencyCode();
-
         $merchantReference = $quote->getReservedOrderId();
         $storeId = $quote->getStoreId();
         $returnUrl = sprintf(
             "%s?merchantReference=%s",
             $this->returnUrlHelper->getStoreReturnUrl($storeId),
             $merchantReference
         );
-
+        $chargedQuoteCurrency = $this->chargedCurrency->getQuoteAmountCurrency($quote);
+        $currency = $chargedQuoteCurrency->getCurrencyCode();
+        $cartSubtotal = $quote->getSubtotal();
```

Comment:
Depending on the value of configuration `payment/adyen_abstract/charged_currency` field, `getBaseSubtotal()` method needs to be used. The main responsibility of `ChargedCurrency` class is providing correct amount (base or display) and currency.

Currency is obtained correctly on line 169 but the amount is obtained directly from quote. This implementation will fail if the `payment/adyen_abstract/charged_currency` config is set to `base`.

Reply:
After our discussion and checking the other express payment methods, plugin ignores `base` currency payments for express payments. In this case, you can ignore my comment above.